tests/validator/validator.cpp

```diff
@@ -1764,3 +1764,36 @@ TEST(Validator, unitComplexCycle)
 
     EXPECT_EQ_ERRORS(expectedErrors, v);
 }
+
+TEST(Validator, unfoundUnitsInEncapsulatedComponents)
+{
+    // This test resulted from https://github.com/cellml/libcellml/issues/430
+    const std::vector<std::string> expectedErrors = {
+        "Variable 'v' has an units reference 'neither_do_I_but_Im_not_reported' that does not correspond with a standard units and is not a units defined in the variable's model.",
+        "Variable 'v' has an units reference 'i_dont_exist' that does not correspond with a standard units and is not a units defined in the variable's model.",
+    };
```

Comment:
I kind of wish there was some logical order in the way errors are reported. Based on the model, I would expect the issue with the `i_dont_exist` unit to be reported first and then the one with the `neither_do_I_but_Im_not_reported` unit since `c3` is a child of `c2`. It's not critical and has nothing to do with this issue as such, just that it would be "cleaner" (a bad word here, I know!).

Otherwise, `neither_do_I_but_Im_not_reported` probably ought to be renamed (since it's now reported...!)?

Reply:
After being on the "other side" while creating the tutorials, I can second that the error messages leave a lot to be desired in terms of usefulness-in-debugging-your-model!  I suggest a separate PR that deals with them all ...

Reply:
It is possible to tidy this up a little it is the order the tree is searched in, currently we are doing a depth first search thus the deeper error gets found first.

Reply:
Yep, exactly.  Also reporting parents of components in their name would be helpful too.

Reply:
> It is possible to tidy this up a little it is the order the tree is searched in, currently we are doing a depth first search thus the deeper error gets found first.

I thought it might have been the case, and I clearly don't think this is the way it should be done.

> Yep, exactly. Also reporting parents of components in their name would be helpful too.

That's indeed what the code generator does when raising issues.

Reply:
I suggest we do a complete overhaul of all the errors, but in a separate PR.